File: src/main/java/com/ouertatani/gestionproduit/model/CommandeDTO.java
package com.ouertatani.gestionproduit.model;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class CommandeDTO {
    private Long id;
    private Long clientId;
    private LocalDateTime dateCommande;
    private String statut;
    private Double total;

    public CommandeDTO(Commande commande) {
        this.id = commande.getId();
        this.clientId = commande.getClientId();
        this.dateCommande = commande.getDateCommande();
        this.statut = commande.getStatut().name();
        this.total = commande.getTotal();
    }
}

File: lib/models/CommandeDTO.dart
class CommandeDTO {
  final int? id;
  final int? clientId;
  final String? dateCommande;
  final String? statut;
  final double? total;

  CommandeDTO({
    this.id,
    this.clientId,
    this.dateCommande,
    this.statut,
    this.total,
  });

  factory CommandeDTO.fromJson(Map<String, dynamic> json) => CommandeDTO(
        id: json['id'],
        clientId: json['clientId'],
        dateCommande: json['dateCommande'],
        statut: json['statut'],
        total: json['total'] != null ? json['total'].toDouble() : null,
      );
}